### PR TITLE
Add fixes to reduce spelling rule false positives

### PR DESCRIPTION
**Pull request by michelle-purcell**
Fixes:

- False positive spelling warnings on Java-related software services/components/products/functions and some acceptable widely used programmatic terms encountered when running vale on Red Hat build of Quarkus docs.
- Removed some duplicate entries in spelling yml.

_Related heading rule enhancements to follow after this is merged._

### CODE REVIEW

**Comment by aireilly** [comment]
Some errors to fix, but this looks like a very useful update!

File: .vale/styles/RedHat/Spelling.yml [hunk 1]
```diff
@@ -384,9 +407,8 @@ filters:
   - Traefik
   - Uber
   - UIDs
-  - URIs?
-  - url
-  - URLs?
+  - URI
+  - URL
```
Comment (aireilly) [inline .vale/styles/RedHat/Spelling.yml:411]: "URLs" and "URIs" acceptable usages aren't they?

Comment (michelle-purcell) [reply .vale/styles/RedHat/Spelling.yml:411]: @aireilly Just to clarify, I removed the plural instances as those are not needed. The expression caters for plural instances of the listed term.
In my tests against this PR, `URLs` and `URIs` are not flagged with a spelling warning.

Also, I did wonder why several items have a question mark against them. Is that a workaround for an issue or a tag to investigate those terms?

Comment (michelle-purcell) [reply .vale/styles/RedHat/Spelling.yml]: @aireilly  Fixed build errors!

Comment (aireilly) [reply .vale/styles/RedHat/Spelling.yml]: the question mark is regex - it means, "one or more of the preceding character".

Comment (michelle-purcell) [reply .vale/styles/RedHat/Spelling.yml]: @aireilly - Doh :face_in_clouds: Thanks for reminding me! I restored those items.

**Review by aireilly — APPROVED** [review]
lgtm

**Pull request merged**